server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4516,6 +4537,35 @@ public void onAfter() {
                     flushOrRollRunning.compareAndSet(true, false);
                 }
             }
+        } else if (shouldRefreshShard() && isRefreshRunning.compareAndSet(false, true)) {
+
+            if (shouldRefreshShard()) {
+                logger.info("submitting async Refresh request");
+                final AbstractRunnable _refresh = new AbstractRunnable() {
+                    @Override
+                    public void onFailure(Exception e) {
+                        logger.warn("forced refresh failed after number of uncommited translog files breached limit", e);
+                    }
+
+                    @Override
+                    protected void doRun() throws Exception {
+                        refresh("Too many uncommited Translog files");
+                    }
+
+                    @Override
+                    public boolean isForceExecution() {
+                        return true;
+                    }
+
+                    @Override
+                    public void onAfter() {
+                        isRefreshRunning.compareAndSet(true, false);
+                    }
+                };
+                threadPool.executor(ThreadPool.Names.REFRESH).execute(_refresh);
```

Comment:
we need to handle cases where an exception is thrown at the time of submitting the task for execution.

Reply:
This is getting handled implicitly by  execute(). 
```  
 public void execute(Runnable command) {
        command = wrapRunnable(command);
        try {
            super.execute(command);
        } catch (OpenSearchRejectedExecutionException ex) {
            if (command instanceof AbstractRunnable) {
                // If we are an abstract runnable we can handle the rejection
                // directly and don't need to rethrow it.
                try {
                    ((AbstractRunnable) command).onRejection(ex);
                } finally {
                    ((AbstractRunnable) command).onAfter();

                }
            } else {
                throw ex;
            }
        }
    }